File: apps/backend/internal/model/category/category.go
package category

import (
	"time"

	"github.com/google/uuid"
)

type CategoryType string

const (
	CategoryTypeIncome  CategoryType = "income"
	CategoryTypeExpense CategoryType = "expense"
)

type Category struct {
	ID        uuid.UUID    `json:"id" db:"id"`
	UserID    string       `json:"user_id" db:"user_id"`
	Name      string       `json:"name" db:"name"`
	Type      CategoryType `json:"type" db:"type"`
	CreatedAt time.Time    `json:"created_at" db:"created_at"`
}

File: packages/zod/src/category.ts
import { z } from "zod";

export const ZCategory = z.object({
  id: z.string().uuid(),
  user_id: z.string(),
  name: z.string().max(100),
  type: z.enum(["income", "expense"]),
  created_at: z.string().datetime(),
});

export const ZCreateCategoryRequest = z.object({
  name: z.string().min(1).max(100),
  type: z.enum(["income", "expense"]),
});

export const ZUpdateCategoryRequest = z.object({
  name: z.string().min(1).max(100),
  type: z.enum(["income", "expense"]),
});
